### ISSUE TITLE
About the answer prediction

### ISSUE THREAD

**Issue by YeDeming** [allenai/document-qa#25]
Hi Christopher,

I still have a small question:

In Class BoundaryPrediction at docqa/nn/span_prediction.py, in function get_best_span(), I find you convert start_logits and end_logits to function best_span_from_bounds().

As far as I am concerned, we should convert tf.nn.log_softmax(start_logits) && tf.nn.log_softmax(end_logits) rather than start_logits&end_logits ?

Thanks a lot for your time!
YeDeming

**Comment by chrisc36** [comment]
You can do either one, since the span with the logit will also be the span the highest probability.

**Comment by YeDeming** [comment]
I think they are not the same? 

In function best_span_from_bounds, we rely on the sum of this two parameters to select answer. If we use tf.nn.log_softmax(start_logits) && tf.nn.log_softmax(end_logits), the meaning of the sum is log(p(start|doc,question)*p(end|doc,question)).

**Comment by chrisc36** [comment]
best_span computes max(log(p(start|doc,question) + log(p(end|doc,question))), which will return the same result as max(p(start|doc,question) * (p(end|doc,question))

**Comment by YeDeming** [comment]
yep, but when we convert the start_logits&end_logits to best_span in function get_best_span(),  the meaning is not max(p(start|doc,question) * p(end|doc,question))
